src/vivarium_csu_ltbi/components/observers.py

```diff
@@ -124,44 +126,29 @@ def __init__(self):
 
     def setup(self, builder):
         super().setup(builder)
-        self.household_tb_exposure = builder.value.get_value('household_tuberculosis.exposure')
+        self.household_tb_exposure = builder.value.get_value(f'{ltbi_globals.HOUSEHOLD_TUBERCULOSIS}.exposure')
 
     def metrics(self, index, metrics):
         pop = self.population_view.get(index)
         pop.loc[pop.exit_time.isnull(), 'exit_time'] = self.clock()
 
         exposure_category = self.household_tb_exposure(index)
 
-        exposed = pop.loc[exposure_category == 'cat1']
-        unexposed = pop.loc[exposure_category == 'cat2']
-
-        exposed_person_time = get_person_time(exposed, self.config.to_dict(), self.start_time, self.clock(),
-                                              self.age_bins)
-        exposed_person_time = {f'{k}_exposed_to_hhtb': v for k, v in exposed_person_time.items()}
-        exposed_deaths = get_deaths(exposed, self.config.to_dict(), self.start_time, self.clock(), self.age_bins,
-                                    self.causes)
-        exposed_deaths = {f'{k}_exposed_to_hhtb': v for k, v in exposed_deaths.items()}
-        exposed_ylls = get_years_of_life_lost(exposed, self.config.to_dict(), self.start_time, self.clock(),
-                                              self.age_bins, self.life_expectancy, self.causes)
-        exposed_ylls = {f'{k}_exposed_to_hhtb': v for k, v in exposed_ylls.items()}
-
-        metrics.update(exposed_person_time)
-        metrics.update(exposed_deaths)
-        metrics.update(exposed_ylls)
-
-        unexposed_person_time = get_person_time(unexposed, self.config.to_dict(), self.start_time, self.clock(),
-                                                self.age_bins)
-        unexposed_person_time = {f'{k}_unexposed_to_hhtb': v for k, v in unexposed_person_time.items()}
-        unexposed_deaths = get_deaths(unexposed, self.config.to_dict(), self.start_time, self.clock(), self.age_bins,
-                                    self.causes)
-        unexposed_deaths = {f'{k}_unexposed_to_hhtb': v for k, v in unexposed_deaths.items()}
-        unexposed_ylls = get_years_of_life_lost(unexposed, self.config.to_dict(), self.start_time, self.clock(),
-                                              self.age_bins, self.life_expectancy, self.causes)
-        unexposed_ylls = {f'{k}_unexposed_to_hhtb': v for k, v in unexposed_ylls.items()}
-
-        metrics.update(unexposed_person_time)
-        metrics.update(unexposed_deaths)
-        metrics.update(unexposed_ylls)
+        measure_getters = (
```

Comment:
This is the only significant refactor in this PR.  Sorry, should have saved it for a separate PR.  No logic changes here, just converting to a loop.